### PR TITLE
Add basic UI for scheduling workflows

### PR DESCRIPTION
**Pull request by tomasol**
Summary:
Add 'Scheduled' tab to workflow UI

New schedules are created in workflow definition tab. Once created, they can be updated or deleted in 'Scheduled' tab. Currently only enabled/disabled and cron string can be edited.

Test plan:
setup:
Start symphony, conductor.frinx, workflows containers.
Create new workflow named `workflow1` e.g. using the builder.
test:
Open workflow definition, expand `workflow1`. Last button in the bar should be `Create schedule` :

Click the button, new modal window should appear:

Click `Submit`. Expanding the `workflow1`, last button should be `Edit schedule`.
Click `Scheduled` tab, all schedules should appear:

Highlight newly created schedule, updating and deleting should work as expected.
If you set the cron expression to `0 * * ? * *`, new execution should appear every minute. Cron expressions are evaluated using [go cron library](https://github.com/robfig/cron) which in turn uses [Quartz expressions](http://www.quartz-scheduler.org/documentation/quartz-2.3.0/tutorials/tutorial-lesson-06.html)

### CODE REVIEW

File: symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        // TODO
+      }
+    });
+    return [];
+  };
+
+  const [data, setData] = useState(refresh);
+  const [pagesCount, setPagesCount] = useState(1);
+  const [defaultPages, setDefaultPages] = useState(20);
+  const [viewedPage, setViewedPage] = useState(1);
```
Comment (KenG98) [inline symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js]: Could these go on line 21 considering that `refresh()`, which is above these, references them?

File: symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        setPagesCount(result.length % defaultPages ? ++size : size);
+        deselectActiveRow();
+      } else {
+        // TODO
+      }
+    });
+    return [];
+  };
+
+  const [data, setData] = useState(refresh);
```
Comment (KenG98) [inline symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js]: Could you clarify what's happening here? refresh is a lambda, how is it used as "state" here? Also, I'm seeing `data` and `setData` initialized here using refresh, but the refresh lambda references `setData` (which seems to me like a circular dependency). Just need some clarity here. Thanks.

**Review by KenG98 — CHANGES_REQUESTED** [review]
Looking forward to using this cool feature. I've added a couple comments. Also, could you add screenshots? Thanks!

File: symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js (hunk 2 above)
Comment (tomasol) [reply symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js]: I've struggled there trying to use [Lazy initial state](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state) however I needed the function result to be a promise and that does not work, so I have returned [] and did the actual state initialization in the lambda.
After your comment I've read an [article](https://codeburst.io/master-the-asynchronous-state-in-a-functional-react-component-60a0899cffca) about async & functional components and modified the initialization according to it. Not sure which approach is preferable, I will be happy to redo it if there is a more idiomatic way.

Comment (KenG98) [reply symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js]: Neat, I read the article and that looks like a good solution to me 👍 thanks, will approve the PR

**Review by KenG98 — APPROVED** [review]

**Pull request closed without merging**